web/api/wrapper.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+// WrapperCommand represents components necessary for OAuth2l request
+type WrapperCommand struct {
+	RequestType string
+	Args
+	Credential
+}
+
+// Args type used for unmarshalled JSON
+type Args map[string]interface{}
+
+// Credential type used for storing JSON-formatted credentials
+type Credential map[string]interface{}
+
+// Execute will capture output of OAuth2l CLI using command args
```

Comment:
nit: First sentence of function comment should focus on the core purpose of the function. After that, include additional details as needed. Suggestion: "Execute invokes OAuth2l CLI with the arguments from WrapperCommand. The output of the CLI is captured and returned as a string."